File: backend/app/schemas/plant.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel, Field

class PlantTaskBase(BaseModel):
    task_type: str  # water, fertilize, repot, mist, clean_leaves, custom
    due_date: str  # YYYY-MM-DD
    interval_days: int = 7
    notes: Optional[str] = None
    is_completed: bool = False

class PlantTaskCreate(PlantTaskBase):
    pass

class PlantTaskResponse(PlantTaskBase):
    id: int
    plant_id: int
    last_completed_at: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class PlantLogEntryBase(BaseModel):
    entry_type: str = "photo"  # photo, note, repotting, ai_diagnosis
    image_url: Optional[str] = None
    title: str
    notes: Optional[str] = None

class PlantLogEntryCreate(PlantLogEntryBase):
    pass

class PlantLogEntryResponse(PlantLogEntryBase):
    id: int
    plant_id: int
    created_at: datetime

    class Config:
        from_attributes = True


class PlantBase(BaseModel):
    name: str
    species_latin: Optional[str] = None
    species_czech: Optional[str] = None
    room: str = "living_room"
    light_requirement: str = "bright_indirect"
    watering_interval_days: int = 7
    winter_watering_interval_days: int = 14
    fertilizing_interval_days: int = 14
    misting_required: bool = False
    pot_diameter_cm: Optional[float] = None
    substrate_type: Optional[str] = None
    pet_toxicity: str = "safe"  # safe, toxic, mildly_toxic
    pet_toxicity_notes: Optional[str] = None
    primary_image_url: Optional[str] = None
    health_status: str = "healthy"  # healthy, needs_attention, sick
    health_notes: Optional[str] = None
    last_watered_date: Optional[str] = None
    next_watering_date: Optional[str] = None
    is_winter_mode: bool = False
    is_favorite: bool = False
    notes: Optional[str] = None

class PlantCreate(PlantBase):
    pass

class PlantUpdate(BaseModel):
    name: Optional[str] = None
    species_latin: Optional[str] = None
    species_czech: Optional[str] = None
    room: Optional[str] = None
    light_requirement: Optional[str] = None
    watering_interval_days: Optional[int] = None
    winter_watering_interval_days: Optional[int] = None
    fertilizing_interval_days: Optional[int] = None
    misting_required: Optional[bool] = None
    pot_diameter_cm: Optional[float] = None
    substrate_type: Optional[str] = None
    pet_toxicity: Optional[str] = None
    pet_toxicity_notes: Optional[str] = None
    primary_image_url: Optional[str] = None
    health_status: Optional[str] = None
    health_notes: Optional[str] = None
    last_watered_date: Optional[str] = None
    next_watering_date: Optional[str] = None
    is_winter_mode: Optional[bool] = None
    is_favorite: Optional[bool] = None
    notes: Optional[str] = None

class PlantResponse(PlantBase):
    id: int
    created_by_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime
    
    # Calculated fields
    days_until_watering: int = 0
    is_thirsty: bool = False
    watering_status: str = "ok"  # ok, due_today, overdue, watered_today
    
    tasks: List[PlantTaskResponse] = []
    log_entries: List[PlantLogEntryResponse] = []

    class Config:
        from_attributes = True


class PlantAiAnalyzeRequest(BaseModel):
    plant_name: Optional[str] = None
    image_base64: Optional[str] = None
    image_url: Optional[str] = None
    target_language: str = "cs"


class PlantAiExtracted(BaseModel):
    common_name: str
    species_latin: str
    species_czech: str
    description: str
    light_requirement: str  # direct_sun, bright_indirect, semi_shade, shade
    watering_interval_days: int
    winter_watering_interval_days: int
    fertilizing_interval_days: int
    misting_required: bool
    substrate_recommendation: str
    pet_toxicity: str  # safe, toxic, mildly_toxic
    pet_toxicity_details: str
    initial_health_assessment: str


class PlantDiagnosisRequest(BaseModel):
    plant_id: Optional[int] = None
    plant_name: Optional[str] = None
    symptoms_description: str
    image_base64: Optional[str] = None
    image_url: Optional[str] = None
    target_language: str = "cs"


class PlantDiagnosisResponse(BaseModel):
    plant_name: str
    diagnosis: str
    cause: str
    severity: str  # low, medium, high
    is_contagious: bool
    action_steps: List[str]
    prevention_tips: str


class PlantSitterScheduleItem(BaseModel):
    plant_id: int
    plant_name: str
    species: str
    room: str
    image_url: Optional[str] = None
    action_required: str  # water, do_not_touch, mist
    recommended_water_amount: str
    instructions: str

File: frontend/src/types/plant.ts
export type RoomType =
  | 'living_room'
  | 'bedroom'
  | 'kitchen'
  | 'bathroom'
  | 'balcony'
  | 'hallway'
  | 'office'
  | 'other';

export type LightRequirement = 'direct_sun' | 'bright_indirect' | 'semi_shade' | 'shade';
export type PetToxicity = 'safe' | 'toxic' | 'mildly_toxic';
export type HealthStatus = 'healthy' | 'needs_attention' | 'sick';
export type WateringStatus = 'ok' | 'due_today' | 'overdue' | 'watered_today';

export interface PlantTask {
  id: number;
  plant_id: number;
  task_type: 'water' | 'fertilize' | 'repot' | 'mist' | 'clean_leaves' | 'custom';
  due_date: string;
  last_completed_at?: string;
  interval_days: number;
  notes?: string;
  is_completed: boolean;
  created_at: string;
}

export interface PlantLogEntry {
  id: number;
  plant_id: number;
  entry_type: 'photo' | 'note' | 'repotting' | 'ai_diagnosis';
  image_url?: string;
  title: string;
  notes?: string;
  created_at: string;
}

export interface Plant {
  id: number;
  name: string;
  species_latin?: string;
  species_czech?: string;
  room: RoomType;
  light_requirement: LightRequirement;
  watering_interval_days: number;
  winter_watering_interval_days: number;
  fertilizing_interval_days: number;
  misting_required: boolean;
  pot_diameter_cm?: number;
  substrate_type?: string;
  pet_toxicity: PetToxicity;
  pet_toxicity_notes?: string;
  primary_image_url?: string;
  health_status: HealthStatus;
  health_notes?: string;
  last_watered_date?: string;
  next_watering_date?: string;
  is_winter_mode: boolean;
  is_favorite: boolean;
  notes?: string;
  created_by_id?: number;
  created_at: string;
  updated_at: string;
  
  // Calculated on server
  days_until_watering: number;
  is_thirsty: boolean;
  watering_status: WateringStatus;
  
  tasks?: PlantTask[];
  log_entries?: PlantLogEntry[];
}

export interface PlantAiExtracted {
  common_name: string;
  species_latin: string;
  species_czech: string;
  description: string;
  light_requirement: LightRequirement;
  watering_interval_days: number;
  winter_watering_interval_days: number;
  fertilizing_interval_days: number;
  misting_required: boolean;
  substrate_recommendation: string;
  pet_toxicity: PetToxicity;
  pet_toxicity_details: string;
  initial_health_assessment: string;
}

export interface PlantDiagnosisResponse {
  plant_name: string;
  diagnosis: string;
  cause: string;
  severity: 'low' | 'medium' | 'high';
  is_contagious: boolean;
  action_steps: string[];
  prevention_tips: string;
}

export interface PlantSitterScheduleItem {
  plant_id: number;
  plant_name: string;
  species: string;
  room: string;
  image_url?: string;
  action_required: 'water' | 'do_not_touch' | 'mist';
  recommended_water_amount: string;
  instructions: string;
}
